lagrange/lagrange.py

```diff
@@ -158,19 +189,27 @@ def interpolate(
     if prime <= 1:
         raise ValueError('expecting a positive integer prime modulus')
 
-    # Compute the Lagrange coefficients at ``0``.
-    coefficients = {}
-    for i in range(1, len(values) + 1):
-        coefficients[i] = 1
-        for j in range(1, len(values) + 1):
-            if j != i:
-                coefficients[i] = (coefficients[i] * (0 - j) * _inv(i - j, prime)) % prime
-
-    value = 0
-    for i in range(1, len(values) + 1):
-        value = (value + values[i] * coefficients[i]) % prime
-
-    return value
```

Comment:
I couldn't figure out how to adapt this code, so instead I rewrote it.  The combination of indexes/x-values/y-values/coefficients is confusing, and I'm pretty sure the y-intercepts of the unscaled basis polynomials are not "coefficients" of anything.